### ISSUE TITLE
[apex] VariableNamingConventions and enums

### ISSUE THREAD

**Issue by cropredyHelix** [pmd/pmd#702]
**Rule Set:**
VariableNamingConventions
`Variables that are final and static should be all capitals, 'Foo' is not all capitals.`

**Description:**
When declaring enums, the enum values are often in lower case, especially if you are trying to match SFDC allowed values for specific SObjects. The rule generates false positives.

**Code Sample demonstrating the issue:**

The enum values here are taken from the allowed values in the [AsyncApexJob SObject](https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_asyncapexjob.htm?search_text=emailtemplate), field `JobType` and said field's values are mixed case.

Suggestion: enums shouldn't be subject to this rule.

```
private enum AsyncApexJobType 	{Future,SharingRecalculation,ScheduledApex,BatchApex, 												
                              BatchApexWorker,TestRequest,TestWorker,ApexToken,Queueable}
```

**Running PMD through:** *[Other: Gearset]*

**Comment by rsoesemann** [comment]
@jsotuyod I see this symptom when I run 6.0.1 on my codebase but I am unable to reproduce this when I add a test to the VariableNaming test xml? Any idea why?

**Comment by jsotuyod** [comment]
@rsoesemann not really, since Apex doesn't rely on compiled code you should see the exact same thing...

**Comment by rsoesemann** [comment]
You mean suppressing this rule? No I don't. Can you reproduce this?

**Comment by jsotuyod** [comment]
I confirm this can't be reproduced. I've just pushed a test case that covers this, and it's passing out of the box.
